stdpopsim/slim_engine.py

```diff
@@ -470,21 +470,36 @@
                     g_start+" > g_end="+g_end);
             }
 
-            community.registerLateEvent(NULL,
-                "{dbg('s="+selection_coeff+", h="+dominance_coeff+
-                " for m"+mut_type+" in p"+pop_id+"');}",
-                g_start, g_start);
-            community.registerLateEvent(NULL,
-                "{dbg('s, h defaults for m"+mut_type+" in p"+pop_id+"');}",
-                g_end, g_end);
             /* We explicitly format() here to prevent integral-valued floats
              * from getting converted to integers during string interpolation
              * (this triggers a type error when the fitness callback runs). */
             f_hom = format("%e", 1 + selection_coeff);
             f_het = format("%e", 1 + selection_coeff * dominance_coeff);
-            sim.registerMutationEffectCallback(NULL,
-                "{if (homozygous) return "+f_hom+"; else return "+f_het+";}",
-                mut_type, pop_id, g_start, g_end);
+
+            /* "All populations" is encoded by a negative value of pop_id. */
```

Comment:
Oh dear, is there a different way to do this? Like `None`? Here it would be easy to make a programming mistake? I realize that this is internal only, so not a big deal, though.

Reply:
Yeah I agree, I think setting pop_id to NULL should do it (and would simplify that code blurb).

Reply:
Ugh, using NULL in place of -1 is more complicated than I thought, because of how data vectors are assembled in `slim_makescript`. Let's pass on this for now.